### PR TITLE
Add test cases of delete and update pymilvus version

### PR DESCRIPTION
**Pull request by NicoYuan1986**
1. modify test case about collection.describe() #27055
2. update pymilvus version #17614
3. Code format optimization
4. modify test case about query iterator pagination #26767
5. add test cases of complex expression delete - part one

### CODE REVIEW

File: tests/python_client/common/common_func.py [hunk 1]
```diff
@@ -948,17 +948,18 @@ def gen_normal_expressions():
...
-        "0 < int64 < 400",
-        "500 <= int64 < 1000",
+        "int64 == 0 || float == 10**2 || (int64 + 1) == 3",
+        "0 <= int64 < 400 and int64 % 100 == 0",
         "200+300 < int64 <= 500+500",
-        "int64 in [300/2, 900%40, -10*30+800, 2048/2%200, (100+200)*2]",
-        "float in [+3**6, 2**10/2]",
-        "(int64 % 100 == 0) && int64 < 500",
-        "float <= 4**5/2 && float > 500-1 && float != 500/2+260",
-        "int64 > 400 && int64 < 200",
```
Comment (binbinlv) [inline tests/python_client/common/common_func.py:959]: this expression not covered by your modified code, should not be deleted.

Comment (NicoYuan1986) [reply tests/python_client/common/common_func.py:959]: > this expression not covered by your modified code, should not be deleted.

I have added the line again.

File: tests/python_client/common/common_func.py [hunk 2]
```diff
@@ -948,17 +948,12 @@ def gen_normal_expressions():
...
-        "500 <= int64 < 1000",
+        "int64 == 0 || float == 10**2 || (int64 + 1) == 3",
+        "0 <= int64 < 400 and int64 % 100 == 0",
         "200+300 < int64 <= 500+500",
-        "int64 in [300/2, 900%40, -10*30+800, 2048/2%200, (100+200)*2]",
-        "float in [+3**6, 2**10/2]",
-        "(int64 % 100 == 0) && int64 < 500",
-        "float <= 4**5/2 && float > 500-1 && float != 500/2+260",
         "int64 > 400 && int64 < 200",
-        "float < -2**8",
```
Comment (binbinlv) [inline tests/python_client/common/common_func.py:960]: This one looks not be covered.

File: tests/python_client/testcases/test_delete.py [hunk 3]
```diff
@@ -1851,3 +1851,110 @@ def test_delete_with_string_field_is_empty(self):
...
+        else:
+            data = cf.gen_default_dataframe_data(nb)
+            data[ct.default_json_field_name] = [{"string": str(i)} for i in range(nb)]
+        collection_w.insert(data)
+        collection_w.load()
+
+        # delete with expressions
+        expression = f"{field_name} {like} '1%'"
+        res = collection_w.delete(expression)[0]
+        assert res.delete_count == 1111
```
Comment (binbinlv) [inline tests/python_client/testcases/test_delete.py:1927]: It is better if it is not a fixed number in test cases, could be abstracted.

**Comment by binbinlv** [comment]
The variable "res" is not defined in the modified test case "test_query_output_vec_field_after_index", should be modified.

File: tests/python_client/testcases/test_search.py [hunk 4]
```diff
@@ -2939,7 +2936,7 @@ def test_search_binary_without_flush(self, metrics, auto_id):
 
     @pytest.mark.tags(CaseLabel.L1)
     @pytest.mark.parametrize("expression", cf.gen_normal_expressions())
-    def test_search_with_expression(self, dim, expression, _async, enable_dynamic_field):
+    def test_search_with_expression1(self, dim, expression, _async, enable_dynamic_field):
```
[the remark this answers is not in the document]
Comment (NicoYuan1986) [reply tests/python_client/testcases/test_search.py:2939]: modified. for debug. forget to change back.

File: tests/python_client/common/common_func.py (hunk 2 above)
Comment (NicoYuan1986) [reply tests/python_client/common/common_func.py:960]: covered.

File: tests/python_client/testcases/test_delete.py (hunk 3 above)
Comment (NicoYuan1986) [reply tests/python_client/testcases/test_delete.py:1927]: I have modified it to random string.

**Pull request merged**